clang/lib/Sema/SemaExprCXX.cpp

```diff
@@ -7981,7 +7969,7 @@ ExprResult Sema::ActOnStartCXXMemberReference(Scope *S, Expr *Base,
     SmallVector<FunctionDecl*, 8> OperatorArrows;
     CTypes.insert(Context.getCanonicalType(BaseType));
 
-    while (BaseType->isRecordType()) {
+    while (BaseType->getAsRecordDecl()) {
```

Comment:
This is a bit odd -- we're getting it as a record declaration and ignoring the result. Why is this change needed?

Reply:
It's essentially a shortcut for `BaseType->isRecordType() || isa<InjectedClassNameType>(BaseType->getCanonicalTypeInternal())`

Reply:
`isa<InjectedClassNameType, RecordType>(BaseType)` seems more clear, WDYT?

Reply:
It would have to be spelled `isa<InjectedClassNameType, RecordType>(BaseType.getCanonicalType())`... but that's unnecessarily verbose IMO (especially since we need this check in multiple places). WDYT?

Reply:
I don't insist on a change, but I also don't think that's needlessly verbose. You can put the canonical type in a local variable to shorten it again.

Generally speaking, calling `getXXX()` and not actually caring about the return value aside from whether it's null or not is a bit of a code smell. We ask folks to replace `foo->getAs<Blah>()` and replace it with `isa<Blah>(foo);` quite often and this is just more of the same.

Reply:
Ah, I see. I'll use `isa<InjectedClassNameType, RecordType>(BaseType.getCanonicalType())` then.

Reply:
Perhaps it would be worth introducing `Type::isCXXRecord` for this check. We do it quite often... @cor3ntin WDYT?

Reply:
Sure, that probably make sense. NOT in this pr though, we want to keep to a minimum what we backport